### PR TITLE
fix(cmd.Save): fix bug that doesn't errors on `-dryrun` saves when there exists a previous version

### LINKED ISSUE #645: add Test that confirms you can do a dry run on a dataset that has a previous version
Currently, you cannot do a dry run on a dataset that has a previous version.

When we do a dry run, we pass an in memory repo to `dsfs.CreateDataset`. This memRepo does not contain any other datasets, and when `dsfs.prepareDataset` attempts to load the previous version of the datasets, we get an error.

* [ ] add test that confirms you can do a dry run on a dataset that has a previous version (which should fail)
* [ ] make sure we close qri-io/dataset#164.
* [ ]`Base.CreateDataset`, we need to accept a pointer to a previous dataset, `prev`, and pass this to `dsfs.CreateDataset`. & adjust tests
* [ ] `actions.SaveDataset` we need to pass `prev`, the previous version of our dataset to `base.CreateDataset` & adjust tests
* [ ] make sure dry run test passes!

### PR DESCRIPTION
**Pull request by ramfox**
closes #645 

will fail until qri-io/datasets#166 is closed

Edit: very detailed commit messages explain exactly what's changed and why

### CODE REVIEW

File: actions/dataset.go [hunk 1]
```diff
@@ -185,8 +183,8 @@ func UpdateDataset(node *p2p.QriNode, ref *repo.DatasetRef, secrets map[string]s
 
 func localUpdate(node *p2p.QriNode, ref *repo.DatasetRef, secrets map[string]string, scriptOut io.Writer, dryRun, pin bool) (res repo.DatasetRef, body cafs.File, err error) {
```
Comment (dustmop) [inline actions/dataset.go]: While you're here, could you add a comment to function explaining what it does.

File: base/dataset_prepare_test.go [hunk 2]
```diff
@@ -11,12 +11,12 @@ func TestPrepareDatasetSave(t *testing.T) {
 	r := newTestRepo(t)
 	ref := addCitiesDataset(t, r)
 
-	_, _, _, err := PrepareDatasetSave(r, ref.Peername, ref.Name)
+	_, _, _, _, err := PrepareDatasetSave(r, ref.Peername, ref.Name)
```
[the remark this answers is not in the document]
Comment (ramfox) [reply base/dataset_prepare_test.go]: Actually, taking this comment and running with it, cause I totally just made sure the tests passed without seeing what was really being tested, added a bunch of test that make sure some of the `PrepareDatasetSave` tests are more specific!

File: base/dataset_prepare.go [hunk 3]
```diff
@@ -17,29 +17,30 @@ import (
 )
 
 // PrepareDatasetSave prepares a set of changes for submission to SaveDataset
```
[the remark this answers is not in the document]
Comment (ramfox) [reply base/dataset_prepare.go]: yes!! that will make this so much clearer for future people!! thanks

File: actions/dataset.go (hunk 1 above)
Comment (ramfox) [reply actions/dataset.go]: hey good looking out cause I actually found a bug while reviewing this. making a node in the comment and filing an issue.

**Pull request merged**